### PR TITLE
4277 flexible parsing

### PR DESCRIPTION
**Pull request by kpethtel**
## Summary

Makes the acheron field parsing more flexible because the VAOS team has continued adding new fields and we should expect that the ordering of values may change.

## Related issue(s)
- department-of-veterans-affairs/va-mobile-team#4277

## What areas of the site does it impact?
Impacts mobile appointment requests.

### CODE REVIEW

**Comment by holdenhinkle** [comment]
@kpethtel Nokogiri was bumped earlier today. Please merge master/rebase so the Code Checks / Linting and Security (push) job can pass.

File: modules/mobile/app/models/mobile/v0/adapters/vaos_v2_appointment.rb [hunk 1]
```diff
@@ -409,23 +409,20 @@ def reason_code_contains_embedded_data?
...
-        def embedded_data_matches
-          # preferred modality is equivalent to "kind" which is returned separately, so it's not used
-          @embedded_data_matches ||= appointment.dig(:reason_code, :text)&.match(
-            /preferred modality:?(.*?)\|phone number:?(.*?)\|email:?(.*?)\|preferred dates:?(.*?)\|reason code:?(.*?)\|comments:?(.*)/
-          )
+        def embedded_data_match(key)
+          match = appointment.dig(:reason_code, :text)&.match(/(^|\|)#{key}:?(.*?)(\||$)/)
+          return nil unless match
+
+          match[2].strip.presence
```
Comment (aherzberg) [inline modules/mobile/app/models/mobile/v0/adapters/vaos_v2_appointment.rb:424]: My regex is a little rusty. what are the contents of the first two matches before we grab the 3rd?

Comment (kpethtel) [reply modules/mobile/app/models/mobile/v0/adapters/vaos_v2_appointment.rb:424]: `match` here is a ruby regex `MatchData` object. The first match is the full captured expression (like: `"|phone number: 3174485062|"`), the second is the delimiter from the first match group (either `|` or `^`), the third is the actual capture, and the fourth match is the second delimiter (either `|` or `$`). It's a bit strange, but that's just how it works.

Comment (aherzberg) [reply modules/mobile/app/models/mobile/v0/adapters/vaos_v2_appointment.rb:424]: got it. makes sense.

**Review by aherzberg — APPROVED** [review]
LGTM

**Pull request merged**